### ISSUE TITLE
`find` returns vue component twice

### ISSUE THREAD

**Issue by pangxie1991** [eddyerburgh/avoriaz#27]
Vue: 2.1.8

here is the code
```html
<!-- Item.vue -->
<template>
    <div class="foo">
    </div>
</template>

<!-- Foo.vue -->
<template>
    <div>
          <Foo v-for="item in list" :item="item"></Foo>
    </div>
</template>

<script>
    export default {
         components: {
              'Foo': Foo
         },
         props: ['list']
    }
</script>
```

in Foo.spec.js
```javascript
//......
it('create Foo', () => {
     mount(Foo, {
         propsData: {
             list: []
         }
     });
     wrapper.vm.list = wrapper.vm.list.concat([{}, {}, {}]);
     console.log(wrapper.find('.foo').length); // will print 6, double the actual value
})
//......
```

I try to debug this, and find that in `findAllVNodes`
```javascript
if (vNode.child) {
    findAllVNodes(vNode.child._vnode, nodes, false, depth);
}
```
`vNode.child._vnode.elm == vNode.elm` return true, so this element will push to array `nodes` twice

**Comment by eddyerburgh** [comment]
Hey, this should be fixed in 1.11.0. 

The problem was that Vue components have 2 vnodes. Before returning the vNodes, avoriaz checked for duplicated by comparing the vNodes with lodash.equals. Now it filters out vnodes that have the same element tag, which should remove any duplicates.

**Comment by pangxie1991** [comment]
thank you! :)
